xblock/mixins.py

```diff
@@ -522,7 +522,9 @@ def _add_field(self, node, field_name, field):
             elem.text = value
             node.insert(0, elem)
         else:
-            node.set(field_name, value)
+            # setting the None in the value for node raises TypeError to avoid this
```

Comment:
@cpennington 
 The ```to_string``` at [here](https://github.com/edx/XBlock/blob/017b46b80e712b1318379912257cf1cc1b68eb0e/xblock/fields.py#L792) call to ```to_json()``` [here](https://github.com/edx/XBlock/blob/017b46b80e712b1318379912257cf1cc1b68eb0e/xblock/fields.py#L563) which did'nt handle the ```None``` case so thats why its not returning string for ```None``` Type.

So should we overwrite  ```to_json``` for [class String](https://github.com/edx/XBlock/blob/017b46b80e712b1318379912257cf1cc1b68eb0e/xblock/fields.py#L773) with handling ```None``` case just like [here](https://github.com/edx/XBlock/blob/017b46b80e712b1318379912257cf1cc1b68eb0e/xblock/fields.py#L845) ? or should we handle ```None``` case at ```to_string``` method ?.